src/Persistence_matrix/include/gudhi/Persistence_matrix/base_pairing.h

```diff
@@ -83,26 +77,36 @@ class Base_pairing
    */
   const barcode_type& get_current_barcode();
 
-  /**
-   * @brief Assign operator.
-   */
-  Base_pairing& operator=(Base_pairing other);
   /**
    * @brief Swap operator.
    */
   friend void swap(Base_pairing& pairing1, Base_pairing& pairing2) {
+    if constexpr (Master_matrix::Option_list::has_removable_columns) {
+      swap(static_cast<Face_position_to_ID_mapper<id_index, pos_index>&>(pairing1),
+           static_cast<Face_position_to_ID_mapper<id_index, pos_index>&>(pairing2));
+    }
     pairing1.barcode_.swap(pairing2.barcode_);
     pairing1.deathToBar_.swap(pairing2.deathToBar_);
+    pairing1.idToPosition_.swap(pairing2.idToPosition_);
     std::swap(pairing1.isReduced_, pairing2.isReduced_);
   }
 
  protected:
   using pos_index = typename Master_matrix::pos_index;
+  using id_index = typename Master_matrix::id_index;
   using dictionary_type = typename Master_matrix::bar_dictionary_type;
   using base_matrix = typename Master_matrix::Boundary_matrix_type;
+  using RUM = typename std::conditional<Master_matrix::Option_list::has_removable_columns,
```

Comment:
Does "RUM" come from RU-matrix Map? I am ok with using initials, but there should be at least one comment that helps find out what they stand for.